src/pages/SupplierRequisitionsPage.js

```diff
@@ -218,6 +251,16 @@ export class SupplierRequisitionsPage extends React.Component {
           onClose={() => this.setState({ isCreatingRequisition: false })}
           title={modalStrings.search_for_the_supplier}
         />
+        {byProgramModalOpen && (
+          <ByProgramModal
+            isOpen={byProgramModalOpen}
+            onConfirm={this.onNewRequisition}
+            onCancel={this.onCancelByProgram}
```

Comment:
For onConfirm and onCancel I think it's better to call `this.setState({ byProgramModalOpen: false })` here, i.e:
```
onConfirm={({supplier, ...requisitionValues}) =>
   this.setState({byProgramModalOpen: false}, () => this.onNewRequisition(supplier, requisitionValues))
}
```
And if you change createRecord to be generic for both normal and program requisition, then onNewRequisition would just be:
```
onNewRequisition = (otherStoreName,  requisitionValues) =>{
...
      requisition = createRecord(database, 'Requisition', currentUser, otherStoreName, requisitionValues);	  
...
}
```

Reply:
I consolidated these to use the same method. Even though it's a probably insignificant performance enhancement I think it's best to stay away from creating new functions in the render method as a general rule/convention? 7c48ac37482d327dc51f0f8275ff91a4484445e5